src/main/java/org/kohsuke/github/GitHubRateLimitChecker.java

```diff
@@ -122,24 +136,26 @@ void checkRateLimit(GitHubClient client, GitHubRequest request) throws IOExcepti
 
     /**
      * Gets the appropriate {@link RateLimitChecker} for a particular url path. Similar to
-     * {@link GHRateLimit#getRecordForUrlPath(String)}.
+     * {@link GHRateLimit#getRecord(RateLimitTarget)}.
      *
-     * @param urlPath
-     *            the url path of the request
-     * @return the {@link RateLimitChecker} for a url path.
+     * @param rateLimitTarget
+     *            the rate limit to check
+     * @return the {@link RateLimitChecker} for a particular target
      */
     @Nonnull
-    private RateLimitChecker selectChecker(@Nonnull String urlPath) {
-        if (urlPath.equals("/rate_limit")) {
+    private RateLimitChecker selectChecker(@Nonnull RateLimitTarget rateLimitTarget) {
```

Comment:
Instead of trying to guess corresponding rate limit checker from the url path, depend on the requester to provide that information.  Much simpler.